From 062c3a6707b22cf7494bbf0ccb8ea6559dc334df Mon Sep 17 00:00:00 2001
From: Alexandre Derumier <alexandre.derumier@groupe-cyllene.com>
Date: Fri, 1 Sep 2023 09:43:05 +0200
Subject: [PATCH] bond: fix mac-address on create with first slave mac

since systemd v241, bond (like bridge), are create with a random mac
instead their first slave.

We already fixing it for bond on reload, but not at create

Than mean that on first reload, we'll always ifdown/ifup interface and change mac.
---
 ifupdown2/addons/bond.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/ifupdown2/addons/bond.py b/ifupdown2/addons/bond.py
index 74d2cab5..5e77e1b0 100644
--- a/ifupdown2/addons/bond.py
+++ b/ifupdown2/addons/bond.py
@@ -884,7 +884,7 @@ def _up(self, ifaceobj, ifaceobj_getfunc=None):
                 ifaceobj_getfunc,
             )
 
-            if not self.bond_mac_mgmt or not link_exists or ifaceobj.get_attr_value_first("hwaddress"):
+            if not self.bond_mac_mgmt or ifaceobj.get_attr_value_first("hwaddress"):
                 return
 
             # check if the bond mac address is correctly inherited from it's
